drivers/interrupt_controller/intc_gic.c

```diff
@@ -11,28 +11,92 @@
  *       GICv3 interface is not supported.
  */
 
-#include <device.h>
 #include <sw_isr_table.h>
-#include <irq_nextlevel.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <drivers/interrupt_controller/gic.h>
 
 #if CONFIG_GIC_VER >= 3
 #error "GICv3 and above are not supported"
 #endif
 
-struct gic_ictl_config {
-	u32_t isr_table_offset;
-};
+void arm_gic_irq_enable(unsigned int irq)
```

Comment:
is it possible to change the name prefixes of the functions in this file?
I'd prefer arm_ to be a prefix for architecture-only functions (ideally, only z_arm)

Although, i know that the ARM MPU internal API uses function prefix arm_core_mpu_ , so this is probably fine.

Reply:
The prefix used here is really `arm_gic_`, not `arm_`; though, this is indeed quite ambiguous and can potentially be misleading.

The rationale for using with `arm_gic_` instead of just `gic_` is as follows:
1. `gic_` alone is relatively short and could potentially be used in the application code for something else.
2. Other interrupt controller driver headers use a similar nomenclature possibly with the same intention (e.g. `sam0_eic_` instead of just `eic_`).
3. Since GIC is often an integral part of the ARM architecture (especially in the ARMv8-A/-R), IMHO, GIC deserves to have `arm` in its prefix.

Reply:
Not super happy about it, but I am convinced that this is probably the best prefix, as it is for MPU functions.
Please, add some documentation that explains that these functions are "often" a part of ARM core implementations (but not strictly architecture natives)

Reply:
I have added some explanations/documentations in `gic.h`.